### ISSUE TITLE
Crash (core dumped) - dependency issues on mbed-client

### ISSUE THREAD

**Issue by mlnx** [ARMmbed/mbed-client-linux-example#16]
The Linux example crashes and shows a serious message:

```
mlnx@mlnx:~/mbed/mbed-client-linux-example$ ./build/x86-linux-native/source/mbed-client-linux-example 
Endpoint Name : a881f383-0819-437a-8a63-7b7fa8f0c6ac

Interface created
register object created
Device object created !!
Registering endpoint

Registered
Incrementing Resource Value
Resource Value /Test/0/D : 1
*** Error in `./build/x86-linux-native/source/mbed-client-linux-example': double free or corruption (fasttop): 0x0000000001488d90 ***
Aborted (core dumped)
```

It seems the latest changes introduced in mbed-client cause the problem
https://github.com/ARMmbed/mbed-client/commit/bea2da2342459f4ae7fd81260c778a228411c086

Tested with the "TechCon-release" branch

```
mlnx@mlnx:~/mbed/mbed-client-linux-example$ yt ls
mbed-client-linux-example 1.0.0
┗━ mbed-client 1.1.12
  ┣━ mbed-client-c 1.0.6 yotta_modules/mbed-client-c
  ┃ ┗━ nanostack-libservice 3.0.7 yotta_modules/nanostack-libservice
  ┣━ mbed-client-linux 1.0.2 yotta_modules/mbed-client-linux
  ┗━ mbed-client-mbedtls 1.0.6 yotta_modules/mbed-client-mbedtls
    ┗━ mbedtls 2.1.4 yotta_modules/mbedtls
```

**Comment by simonqhughes** [comment]
ARM Internal Ref: IOTCLT-372

**Comment by yogpan01** [comment]
Defect fixed and released in mbed-client version 1.1.13. Please verify.

**Comment by mlnx** [comment]
Fixed verified. Works fine now. Thanks.
